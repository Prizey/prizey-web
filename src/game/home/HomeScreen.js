import React from 'react'
import { List } from 'croods'
import { Link } from '@reach/router'
import { Button, Typography } from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles'

import Layout from 'design/Layout/Layout'
import UserBalance from 'design/UserBalance'
import ProfileLink from 'design/ProfileLink/ProfileLink'
import Roulette from '../Roulette'
import SpeedComponent from '../SpeedComponent'

const nextUrl = '/game'

const styles = theme => ({
  login: {
    align: 'left',
    marginRight: '20px',
    [theme.breakpoints.down('xs')]: {
      marginRight: 0,
    },
  },
  root: {
    display: 'block',
    marginTop: parseInt(theme.spacing.lg, 10) * -0.5,
    textDecoration: 'none',
    [theme.breakpoints.down('xs')]: {
      marginTop: 0,
    },
  },
  typography: {
    marginBottom: theme.spacing.md,
    marginTop: theme.spacing.md,
    [theme.breakpoints.down('xs')]: {
      marginBottom: 0,
      marginTop: theme.spacing.xs,
    },
  },
})

const ScreenWithRoullette = withStyles(styles)(
  ({ classes, speed, list, multiplier, currentUser, location }) => (
    <Layout
      leftIcon={<UserBalance />}
      rightIcon={
        currentUser ? (
          <ProfileLink />
        ) : (
          <Link to={'/sign-in?next=/game'} style={{ textDecoration: 'none' }}>
            <Typography className={classes.login}>Login</Typography>
          </Link>
        )
      }
      location={location}
      currentUser={currentUser}
    >
      <Link to={nextUrl} className={classes.root}>
        <Roulette
          aria-label="roulette"
          speed={speed}
          data={list}
          multiplier={multiplier}
        />
        <Typography align="center" variant="h5" className={classes.typography}>
          Tap the screen, <br />
          win what you tap.
        </Typography>
        <Button
          variant="contained"
          color="primary"
          fullWidth
          aria-label="I want it"
        >
          PLAY FOR $1
        </Button>
      </Link>
    </Layout>
  ),
)

const ScreenWithSpeed = (
  difficulty,
  navigate,
  currentUser,
  location,
) => list => (
  <SpeedComponent
    render={settings => {
      const speed = settings[`${difficulty}CarouselSpeed`]

      return (
        <ScreenWithRoullette
          list={list}
          difficulty={difficulty}
          multiplier={settings.priceMultiplier}
          navigate={navigate}
          speed={speed}
          currentUser={currentUser}
          location={location}
        />
      )
    }}
  />
)

export default ({ currentUser, location, navigate, difficulty = 'medium' }) => (
  <List
    currentUser={currentUser}
    location={location}
    name="products"
    path={`/products/homepage`}
    render={ScreenWithSpeed(difficulty, navigate, currentUser, location)}
  />
)
