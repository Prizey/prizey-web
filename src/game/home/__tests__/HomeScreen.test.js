import React from 'react'
import renderer from 'react-test-renderer'
import HomeScreen from '../HomeScreen'

const mockList = [
  { id: 1, image: '/mocks/trump-mask.png', title: 'trump' },
  { id: 2, image: '/mocks/sweatshirt.png', title: 'sweatshirt' },
  { id: 3, image: '/mocks/shoe.png', title: 'shoe' },
]

jest.mock('react-redux', () => ({
  connect: () => Component => props => (
    <Component {...props} chooseProduct={jest.fn} />
  ),
}))

jest.mock('croods', () => ({
  Info: ({ children, ...props }) => (
    <div {...props}>
      Info -{' '}
      {props.render({
        mediumCarouselSpeed: 500,
        priceMultiplier: 1,
      })}
    </div>
  ),
  List: props => (
    <div {...props}>
      {props.children} - Render: {props.render(mockList)}
    </div>
  ),
  Provider: ({ children, ...props }) => (
    <div {...props}>Provider - {children}</div>
  ),
  createReducer: () => (state = {}) => state,
}))

jest.mock('../../../design/Layout/RegisterPageView')

describe('when component is mounted', () => {
  describe('when user is logged in', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(<HomeScreen nextUrl="foo" currentUser />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders correctly with external nextUrl link', () => {
      const tree = renderer
        .create(<HomeScreen nextUrl="foo" currentUser isNextUrlExternal />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('when user is not logged in', () => {
    it('redirects correctly', () => {
      const tree = renderer.create(<HomeScreen nextUrl="foo" />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('redirects correctly with external nextUrl link', () => {
      const tree = renderer
        .create(<HomeScreen nextUrl="foo" isNextUrlExternal />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
