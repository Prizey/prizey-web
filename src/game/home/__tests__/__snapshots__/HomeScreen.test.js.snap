// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when component is mounted when user is logged in renders correctly 1`] = `
<div
  name="products"
  path="/products/homepage"
  render={[Function]}
>
   - Render: 
  <div
    id={1}
    name="game"
    path="/game_setting"
    render={[Function]}
  >
    Info -
     
    <div
      className="Component-root-9"
    >
      <div
        className="Component-header-7"
      />
      <div
        className="Component-container-4"
      >
        <img
          alt="Prizey"
          className="Component-logoImage-8"
          src="/logo.png"
        />
        <div
          className="Component-caption-2"
        />
        <div
          className="Component-children-3"
        >
          <a
            className="Component-root-1"
            href="/game"
            onClick={[Function]}
          >
            <div
              className="Component-root-11"
            >
              <div
                aria-label="rouletteItem-0"
                onMouseDown={[Function]}
              >
                <img
                  alt="trump"
                  className="Component-img-10"
                  pose="enter"
                  speed={NaN}
                  src="/mocks/trump-mask.png"
                />
              </div>
              <div
                aria-label="rouletteItem-1"
                onMouseDown={[Function]}
              >
                <img
                  alt="sweatshirt"
                  className="Component-img-10"
                  pose="leave"
                  speed={NaN}
                  src="/mocks/sweatshirt.png"
                />
              </div>
              <div
                aria-label="rouletteItem-2"
                onMouseDown={[Function]}
              >
                <img
                  alt="shoe"
                  className="Component-img-10"
                  pose="leave"
                  speed={NaN}
                  src="/mocks/shoe.png"
                />
              </div>
            </div>
            <br />
            <h5
              className="MuiTypography-root-12 MuiTypography-h5-28 MuiTypography-alignCenter-35"
            >
              Tap the screen, 
              <br />
              win what you tap.
            </h5>
            <button
              aria-label="I want it"
              className="MuiButtonBase-root-74 MuiButton-root-48 MuiButton-contained-59 MuiButton-containedPrimary-60 MuiButton-raised-62 MuiButton-raisedPrimary-63 MuiButton-fullWidth-73"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-49"
              >
                PLAY FOR $1
              </span>
              <span
                className="MuiTouchRipple-root-77"
              />
            </button>
          </a>
        </div>
        <div
          className="Component-footer-5"
        >
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-12 MuiTypography-caption-22 MuiTypography-colorTextSecondary-45 MuiTypography-alignCenter-35"
            >
              TERMS
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-12 MuiTypography-caption-22 MuiTypography-colorTextSecondary-45 MuiTypography-alignCenter-35"
            >
              PRIVACY
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-12 MuiTypography-caption-22 MuiTypography-colorTextSecondary-45 MuiTypography-alignCenter-35"
            >
              FAIRNESS
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-12 MuiTypography-caption-22 MuiTypography-colorTextSecondary-45 MuiTypography-alignCenter-35"
            >
              ALL PRIZES
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`when component is mounted when user is not logged in redirects correctly 1`] = `
<div
  name="products"
  path="/products/homepage"
  render={[Function]}
>
   - Render: 
  <div
    id={1}
    name="game"
    path="/game_setting"
    render={[Function]}
  >
    Info -
     
    <div
      className="Component-root-9"
    >
      <div
        className="Component-header-7"
      />
      <div
        className="Component-container-4"
      >
        <img
          alt="Prizey"
          className="Component-logoImage-8"
          src="/logo.png"
        />
        <div
          className="Component-caption-2"
        />
        <div
          className="Component-children-3"
        >
          <a
            className="Component-root-1"
            href="/game"
            onClick={[Function]}
          >
            <div
              className="Component-root-11"
            >
              <div
                aria-label="rouletteItem-0"
                onMouseDown={[Function]}
              >
                <img
                  alt="trump"
                  className="Component-img-10"
                  pose="enter"
                  speed={NaN}
                  src="/mocks/trump-mask.png"
                />
              </div>
              <div
                aria-label="rouletteItem-1"
                onMouseDown={[Function]}
              >
                <img
                  alt="sweatshirt"
                  className="Component-img-10"
                  pose="leave"
                  speed={NaN}
                  src="/mocks/sweatshirt.png"
                />
              </div>
              <div
                aria-label="rouletteItem-2"
                onMouseDown={[Function]}
              >
                <img
                  alt="shoe"
                  className="Component-img-10"
                  pose="leave"
                  speed={NaN}
                  src="/mocks/shoe.png"
                />
              </div>
            </div>
            <br />
            <h5
              className="MuiTypography-root-12 MuiTypography-h5-28 MuiTypography-alignCenter-35"
            >
              Tap the screen, 
              <br />
              win what you tap.
            </h5>
            <button
              aria-label="I want it"
              className="MuiButtonBase-root-74 MuiButton-root-48 MuiButton-contained-59 MuiButton-containedPrimary-60 MuiButton-raised-62 MuiButton-raisedPrimary-63 MuiButton-fullWidth-73"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-49"
              >
                PLAY FOR $1
              </span>
              <span
                className="MuiTouchRipple-root-77"
              />
            </button>
          </a>
        </div>
        <div
          className="Component-footer-5"
        >
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-12 MuiTypography-caption-22 MuiTypography-colorTextSecondary-45 MuiTypography-alignCenter-35"
            >
              TERMS
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-12 MuiTypography-caption-22 MuiTypography-colorTextSecondary-45 MuiTypography-alignCenter-35"
            >
              PRIVACY
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-12 MuiTypography-caption-22 MuiTypography-colorTextSecondary-45 MuiTypography-alignCenter-35"
            >
              FAIRNESS
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-12 MuiTypography-caption-22 MuiTypography-colorTextSecondary-45 MuiTypography-alignCenter-35"
            >
              ALL PRIZES
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
